spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkPositionDeltaWrite.java

```diff
@@ -182,15 +183,20 @@ public DeltaWriterFactory createBatchWriterFactory(PhysicalWriteInfo info) {
     }
 
     private Broadcast<Map<String, DeleteFileSet>> broadcastRewritableDeletes() {
-      if (context.deleteGranularity() == DeleteGranularity.FILE && scan != null) {
-        Map<String, DeleteFileSet> rewritableDeletes = scan.rewritableDeletes();
+      if (scan != null && shouldRewriteDeletes()) {
+        Map<String, DeleteFileSet> rewritableDeletes = scan.rewritableDeletes(context.useDVs());
         if (rewritableDeletes != null && !rewritableDeletes.isEmpty()) {
           return sparkContext.broadcast(rewritableDeletes);
         }
       }
       return null;
     }
 
+    private boolean shouldRewriteDeletes() {
+      // Rewritable deletes can either be DVs or file scoped position deletes
```

Comment:
Minor: Other comments seem to start with lower case letters.

Reply:
I think a more accurate description would be that we have to rewrite deletes when dealing with DVs and file-scoped deletes.